src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/MetaGuidanceServiceTests.cs

```diff
@@ -0,0 +1,536 @@
+                Assert.Equal("subject1_indicator", version2Result.Right.Subjects[0].Variables[1].Value);
+
+                Assert.Equal(subject2.Id, version2Result.Right.Subjects[1].Id);
+                Assert.Equal("Version 2 Subject 2 Meta Guidance", version2Result.Right.Subjects[1].Content);
+                Assert.Equal("file2.csv", version2Result.Right.Subjects[1].Filename);
+                Assert.Equal("Subject 2", version2Result.Right.Subjects[1].Name);
+                Assert.Equal("2020_T3", version2Result.Right.Subjects[1].TimePeriods.From);
+                Assert.Equal("2021_T2", version2Result.Right.Subjects[1].TimePeriods.To);
+                Assert.Equal(new List<GeographicLevel>
+                {
+                    GeographicLevel.Country, GeographicLevel.Region
+                }, version2Result.Right.Subjects[1].GeographicLevels);
+                Assert.Equal(2, version2Result.Right.Subjects[1].Variables.Count);
+                Assert.Equal("Subject 2 Filter", version2Result.Right.Subjects[1].Variables[0].Label);
+                Assert.Equal("subject2_filter", version2Result.Right.Subjects[1].Variables[0].Value);
+                Assert.Equal("Subject 2 Indicator", version2Result.Right.Subjects[1].Variables[1].Label);
+                Assert.Equal("subject2_indicator", version2Result.Right.Subjects[1].Variables[1].Value);
+            }
+        }
+
+        [Fact]
+        public async Task UpdateRelease()
+        {
+            var release = new Release
+            {
+                Id = Guid.NewGuid(),
+                PreviousVersionId = null,
+                MetaGuidance = "Release Meta Guidance"
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            var statisticsDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                var service = SetupMetaGuidanceService(contentDbContext: contentDbContext,
+                    statisticsDbContext: statisticsDbContext);
+
+                var result = await service.UpdateRelease(release.Id, new MetaGuidanceUpdateReleaseViewModel
+                {
+                    Content = "Updated Release Meta Guidance"
+                });
+
+                Assert.True(result.IsRight);
+
+                Assert.Equal(release.Id, result.Right.Id);
+                Assert.Equal("Updated Release Meta Guidance", result.Right.Content);
+                Assert.Empty(result.Right.Subjects);
```

Comment:
Could do some more assertions to test that the database has saved the release correctly.

Reply:
Yep, sure, but interestingly we actually need to create the content database context for the third time here to assert this correctly!

If I take away the ` _contentDbContext.Update` and the `await _contentDbContext.SaveChangesAsync()`, as it stands, the test incorrectly passed because as you mentioned I don't assert the database has saved the Release.

BUT

Adding all three of these assertions would still pass, unless they are wrapped to use a new database context, presumably because the entity is already tracked and it's just returned without actually contacting the database. The doc for `FindAsync` mentions this is true.

```
                Assert.Equal("Updated Release Meta Guidance", (await contentDbContext.Releases.FindAsync(release.Id)).MetaGuidance);
                Assert.Equal("Updated Release Meta Guidance", (await contentDbContext.Releases.FirstAsync(r => r.Id == release.Id)).MetaGuidance);
                Assert.Equal("Updated Release Meta Guidance", (await contentDbContext.Releases.Where(r => r.Id == release.Id).FirstAsync()).MetaGuidance);
```

Reply:
Makes sense. Guess it's not really a big deal to create the third context though. Should probably correct any tests that are suspectible to this, as you find them!